### PR TITLE
Item search

### PR DESCRIPTION
**Pull request by HappyNimbus**
All previous issues fixed. @dokdek Please review

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -51,6 +53,24 @@ app.post(
...
+    '/searchItemName/:item_name',
+    async (
+        req: Request,
+        res: Response
+    ) => {
+        try{
+            
+            const response = await searchItemName(req.params.item_name);
+
+                res.send(response);
```
Comment (dokdek) [inline index.ts]: Have you tested this returns a properly formed JSON? Usually its res.json(response), but if this works then thats fine too.

File: searchItemName.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+): Promise<itemDBrow> {
+
+    return new Promise((resolve,reject) => {
+        db.all(
+        
+            "SELECT * FROM items WHERE item_name LIKE'%"+item_name+"%'" ,
+            
+        
+            (error: any, row: any) => {
```
Comment (dokdek) [inline searchItemName.ts]: try not to use `any` in TS, it kind of defeats the purpose of it, you can instead define the row as `row: itemDbRow[]` (or whatever your type is)

File: types/interfaces.ts
Comment (dokdek) [inline types/interfaces.ts:1]: I'll give you my itemDB offline and you can create a type from that, it should be final, so we can sync our types up

File: index.ts [hunk 3]
```diff
@@ -28,20 +29,21 @@ app.get("/testAuth", authenticateToken, (req: Request, res: Response) => {
 app.post(
     "/itemAdd",
     async (
-        req: Request<unknown, unknown, itemStuff, unknown>,
```
Comment (dokdek) [inline index.ts:31]: Re-iterating, I'll give you my itemDB file and we can update this to be relatively final

File: dist/AddItem.js
Comment (dokdek) [inline dist/AddItem.js:1]: Please add your dist folder to `.gitignore` we don't need the compiled files

**Review by dokdek — CHANGES_REQUESTED** [review]
Looks good, just need to update your `.gitignore` so you aren't including your dist folder. Update the types and function signatures to be consistent with the other backend services, after that, should be good to go.

File: index.ts (hunk 1 above)
Comment (HappyNimbus) [reply index.ts]: send works fine, but i will change to json to make things easier.

**Pull request closed without merging**